src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,172 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+    private static final String MESSAGE_INDEX_ERROR = "Index outside the size of the list";
+    private static final String MESSAGE_NO_SUCH_ELEMENT_ERROR = "Element doesn`t exist";
+
+    private Object[] elementData = new Object[DEFAULT_CAPACITY];
+    private int size;
 
     @Override
     public void add(T value) {
+        add(value, elementData, size);
+    }
 
+    private void add(T value, Object[] elementData, int s) {
```

Comment:
s - isn't informative variable name. And do we need this parameter? we can call `elementData.size()` or use size variable instead